src/ui/components/Rating/index.js

```diff
@@ -46,6 +51,34 @@ export class RatingBase extends React.Component {
     // Accept falsey values as if they are zeroes.
     const rating = this.props.rating || 0;
 
+    // We render read-only star ratings with half-stars, to give users a
+    // better representation of the average rating.
+    if (readOnly) {
```

Comment:
Can you not merge this into the non-read-only code? There is a lot of repetition here which will make it hard to maintain.

Also, I think half-stars should also be rendered in non-read-only mode.